sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -251,4 +338,53 @@ class TransformWithStateInPandasStateServer(
       sendResponse(1, s"state $stateName already exists")
     }
   }
+
+  /** Utils object for sending response to Python client. */
+  private object PythonResponseWriterUtils {
+    def sendResponse(
+        status: Int,
+        errorMessage: String = null,
+        byteString: ByteString = null): Unit = {
+      val responseMessageBuilder = StateResponse.newBuilder().setStatusCode(status)
+      if (status != 0 && errorMessage != null) {
+        responseMessageBuilder.setErrorMessage(errorMessage)
+      }
+      if (byteString != null) {
+        responseMessageBuilder.setValue(byteString)
+      }
+      val responseMessage = responseMessageBuilder.build()
+      val responseMessageBytes = responseMessage.toByteArray
+      val byteLength = responseMessageBytes.length
+      outputStream.writeInt(byteLength)
+      outputStream.write(responseMessageBytes)
+    }
+
+    def serializeLongToByteString(longValue: Long): ByteString = {
```

Comment:
I think it may bring some extra complexity to do serde between long and ByteString. Since this is only used in `TimerValueRequests`, maybe we could add a dedicated response message for it which returns a long value? That way we can just use `read_long` on python side.

Reply:
Add a new type of StateResponse to transmit Long type directly in proto message.